subxt/examples/reconnecting_rpc_client.rs

```diff
@@ -0,0 +1,103 @@
+        // This is just a way to ensure that the connection isn't idle if no message is sent that often
+        .enable_ws_ping(
+            PingConfig::new()
+                .ping_interval(Duration::from_secs(6))
+                .inactive_limit(Duration::from_secs(30)),
+        )
+        // There are other configurations as well that can be found here:
+        // <https://docs.rs/reconnecting-jsonrpsee-ws-client/latest/reconnecting_jsonrpsee_ws_client/struct.ClientBuilder.html>
+        .build("ws://localhost:9944".to_string())
+        .await?;
+
+    let api: OnlineClient<PolkadotConfig> =
+        OnlineClient::from_rpc_client(RpcClient::new(rpc.clone())).await?;
+
+    // Build a balance transfer extrinsic.
+    let dest = dev::bob().public_key().into();
+    let balance_transfer_tx = polkadot::tx().balances().transfer_allow_death(dest, 10_000);
+
+    // Submit the balance transfer extrinsic from Alice, and wait for it to be successful
+    // and in a finalized block. We get back the extrinsic events if all is well.
+    let from = dev::alice();
+
+    let mut balance_transfer_progress = api
+        .tx()
+        .sign_and_submit_then_watch_default(&balance_transfer_tx, &from)
+        .await?;
+
+    // When the connection is lost, a error RpcError::DisconnectWillReconnect is emitted on the stream.
+    // in such scenarios the error will be seen here.
+    //
+    // In such scenario it's possible that messages are lost when reconnecting
+    // and if that's acceptable you may just ignore that error message.
+    while let Some(status) = balance_transfer_progress.next().await {
+        match status {
+            // It's finalized in a block!
+            Ok(TxStatus::InFinalizedBlock(in_block)) => {
+                // grab the events and fail if no ExtrinsicSuccess event seen:
+                let events = in_block.wait_for_success().await?;
+                // We can look for events (this uses the static interface; we can also iterate
+                // over them and dynamically decode them):
+                let transfer_event = events.find_first::<polkadot::balances::events::Transfer>()?;
+
+                if transfer_event.is_some() {
+                    println!("Balance transfer success");
+                } else {
+                    println!("Failed to find Balances::Transfer Event");
+                }
+            }
+            // Just log any other status we encounter:
+            //
+            // In this example we emit some important status handling for
+            // here such as Dropped, Invalid etc....
+            Ok(_) => {
+                println!("New status");
+            }
+            // In this example we just ignore when reconnections occurs
+            // but it's technically possible that we can lose
+            // messages on the subscription such as `InFinalizedBlock`
+            // when reconnecting.
+            Err(Error::Rpc(RpcError::DisconnectedWillReconnect(e))) => {
```

Comment:
```suggestion
            Err(e) if e.is_disconnected_will_reconnect() => {
```

I guess something like this is an option too, but the matching is sortof clearer in this example!